### PR TITLE
add random allocator test for continue_add

### PR DESCRIPTION
**Pull request by LHT129**
- implement for hgraph and hnsw

### CODE REVIEW

File: tests/fixtures/random_allocator.h [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    void
+    Deallocate(void* p) override {
+        return free(p);
+    }
+
+    void*
+    Reallocate(void* p, size_t size) override {
+        return realloc(p, size);
```
Comment (jiaweizone) [inline tests/fixtures/random_allocator.h:60]: Reallocate also need a allocate failure case ?

File: tests/fixtures/random_allocator.h [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    RandomAllocator() {
+        rd_ = std::make_shared<std::random_device>();
+        gen_ = std::make_shared<std::mt19937>((*rd_)());
+        std::uniform_int_distribution<int> seed_random;
+        std::uniform_real_distribution<float> ratio_range(0.001f, 0.1f);
+        int seed = seed_random(*gen_);
+        float r = ratio_range(*gen_);
+        std::cout << "seed: " << seed << "   error ratio: " << r << std::endl;
+        gen_->seed(seed);
+        error_ratio_ = r;
```
Comment (jiaweizone) [inline tests/fixtures/random_allocator.h]: Using a fixed constant for `error_ratio_` would be more reliable/definite in identifying problems

File: tests/fixtures/random_allocator.h (hunk 1 above)
Comment (LHT129) [reply tests/fixtures/random_allocator.h:60]: done

File: tests/fixtures/random_allocator.h (hunk 2 above)
Comment (LHT129) [reply tests/fixtures/random_allocator.h]: done

File: src/data_cell/flatten_datacell.h [hunk 3]
```diff
@@ -166,14 +166,13 @@ FlattenDataCell<QuantTmpl, IOTmpl>::BatchInsertVector(const float* vectors,
                                                       InnerIdType count,
                                                       InnerIdType* idx) {
     if (idx == nullptr) {
-        auto* codes = reinterpret_cast<uint8_t*>(
-            allocator_->Allocate(static_cast<uint64_t>(count) * static_cast<uint64_t>(code_size_)));
-        quantizer_->EncodeBatch(vectors, codes, count);
-        io_->Write(codes,
+        Vector<uint8_t> codes(static_cast<uint64_t>(count) * static_cast<uint64_t>(code_size_),
```
Comment (inabao) [inline src/data_cell/flatten_datacell.h]: This will repeatedly call memory allocation for memory allocation.

Comment (LHT129) [reply src/data_cell/flatten_datacell.h]: done

**Review by jiaweizone — APPROVED** [review]
LGTM

**Review by inabao — APPROVED** [review]
LGTM

**Review by wxyucs — APPROVED** [review]
lgtm

**Review by ShawnShawnYou — APPROVED** [review]

**Pull request merged**